### PR TITLE
qa: Replace array by data providers in unit tests for the hostname validator

### PR DESCRIPTION
**Pull request by codisart**
Signed-off-by: codisart <louis.celier@gmail.com>

|    Q          |   A
|-------------- | ------
| QA            | yes

### Description

Replace array for configuration of tests by data providers.
It allows a better isolation of failing tests

### CODE REVIEW

File: test/HostnameTest.php [hunk 1]
```diff
@@ -45,11 +45,20 @@ protected function tearDown() : void
     /**
      * Ensures that the validator follows expected behavior
      *
+     * @dataProvider basicDataProvider
      * @return void
      */
-    public function testBasic()
+    public function testBasic($option, $expected, $inputValues)
```
Comment (weierophinney) [inline test/HostnameTest.php]: Please add `@param` annotations for these new arguments.

File: test/HostnameTest.php [hunk 2]
```diff
@@ -45,11 +45,20 @@ protected function tearDown() : void
...
+        $validator = new Hostname($option);
+        foreach ($inputValues as $input) {
+            $this->assertSame($expected, $validator->isValid($input));
+        }
+    }
+
+    public function basicDataProvider()
+    {
+        return [
             [Hostname::ALLOW_IP, true, ['1.2.3.4', '10.0.0.1', '255.255.255.255']],
```
Comment (weierophinney) [inline test/HostnameTest.php]: I'd argue that you should split these out, so we can have failures rise for each of the IP/hostname arguments; that will allow us to identify specific recursions later more easily.

Additionally, provide names for each case:

```php
'allow-ip succeeds for 1.2.3.4' => [Hostname::ALLOW_IP, true, '1.2.3.4'],
```

As noted in a PR you've made against laminas-validator, this helps us identify specific data sets more easily when they fail.

**Review by weierophinney — CHANGES_REQUESTED** [review]
Thanks for these improvements! I've add a couple notes for further improvement; while they're at specific locations, they apply generally to your entire patch.

File: test/HostnameTest.php (hunk 1 above)
Comment (codisart) [reply test/HostnameTest.php]: I have added the annotations

File: test/HostnameTest.php (hunk 2 above)
Comment (codisart) [reply test/HostnameTest.php]: I have added the keys and separated the test cases

**Pull request merged**